internal/sink/matcher.go

```diff
@@ -194,3 +205,76 @@ func (m1 *MatcherV1) getUrls() (urls *ring.Ring) {
 	m1.urls = m1.urls.Next()
 	return
 }
+
+// Update applies user configurable values for the outbound sender when a
+// webhook is registered
+func (m2 *MatcherV2) update(l ancla.RegistryV2) error {
+
+	//TODO: don't believe the logger for webhook is being set anywhere just yet
+	m2.logger = m2.logger.With(zap.String("webhook.address", l.Registration.Address))
+
+	if l.Registration.FailureURL != "" {
+		_, err := url.ParseRequestURI(l.Registration.FailureURL)
+		if err != nil {
+			return err
+		}
+	}
+
+	//TODO: should we be checking that the l.Registration.Matcher.Field is a field in the wrp.Message?
+	matcher := make(map[string]*regexp.Regexp)
+	for _, item := range l.Registration.Matcher {
+		if item.Regex == ".*" {
+			// Match everything - skip the filtering
+			matcher[item.Field] = &regexp.Regexp{}
+			break
+		}
+
+		var re *regexp.Regexp
+		re, err := regexp.Compile(item.Regex)
+		if err != nil {
+			return fmt.Errorf("invalid matcher item: '%s'", item.Regex)
+		}
+		matcher[item.Field] = re
+	}
+
+	// write/update sink sender
+	m2.mutex.Lock()
+	defer m2.mutex.Unlock()
+
+	// if matcher list is empty set it nil for Queue() logic
+	if len(matcher) == 0 {
+		m2.matcher = nil
+	} else {
+		m2.matcher = matcher
+	}
+
+	return nil
+
+}
+
+func (m2 *MatcherV2) IsMatch(msg *wrp.Message) bool {
+	m2.mutex.RLock()
+	matcher := m2.matcher
+	m2.mutex.RUnlock()
+
+	var (
+		matchDevice = false
+	)
+	for field, deviceRegex := range matcher {
+		if deviceRegex.MatchString(field) || deviceRegex.MatchString(strings.TrimPrefix(field, "event:")) {
+			matchDevice = true
+			break
+		}
+	}
+
+	if !matchDevice {
+		m2.logger.Debug("device regex doesn't match", zap.String("event.source", msg.Source))
+		return false
+	}
+	return true
+}
+
+// TODO: this is a big reason why I want to refactor the Matcher logic
+func (m2 *MatcherV2) getUrls() (urls *ring.Ring) {
```

Comment:
I don't believe MatcherV2 is going to need getUrls - in my refactored branch i move the urls to the matcher to get rid of this function which is also called here: https://github.com/xmidt-org/caduceus/blob/dad38f9d4e160aedc223f6cac33c67289b341d66/internal/sink/sinkSender.go#L423 and why I originally had getUrls a part of the interface